src/comestible_inv.cpp

```diff
@@ -0,0 +1,780 @@
+            }
+            // Might have changed a stack (activated an item, repaired an item, etc.)
+            if( pane->get_area()->info.id == ai::AIM_INVENTORY ) {
+                g->u.inv.restack( g->u );
+            }
+            refresh( true, redraw );
+        } else {
+            item &it = *sitem->items.front();
+            std::vector<iteminfo> vThisItem;
+            std::vector<iteminfo> vDummy;
+            it.info( true, vThisItem );
+            int iDummySelect = 0;
+            ret = draw_item_info( info_startx,
+                                  info_width, 0, 0, it.tname(), it.type_name(), vThisItem, vDummy, iDummySelect,
+                                  false, false, true ).get_first_input();
+        }
+        if( ret == KEY_NPAGE || ret == KEY_DOWN ) {
+            pane->scroll_by( +1 );
+        } else if( ret == KEY_PPAGE || ret == KEY_UP ) {
+            pane->scroll_by( -1 );
+        }
+        refresh( recalc, true ); // item info window overwrote the other pane and the header
+    } else if( action == "QUIT" ) {
+        exit = true;
+    } else if( action == "PAGE_DOWN" ) {
+        pane->scroll_by( +pane->itemsPerPage );
+    } else if( action == "PAGE_UP" ) {
+        pane->scroll_by( -pane->itemsPerPage );
+    } else if( action == "DOWN" ) {
+        pane->scroll_by( +1 );
+    } else if( action == "UP" ) {
+        pane->scroll_by( -1 );
+    } else if( action == "TOGGLE_VEH" ) {
+        if( pane->get_area()->has_vehicle() ) {
+            if( pane->get_area()->info.id != ai::AIM_DRAGGED ) {
+                // Toggle between vehicle and ground
+                pane->set_area( pane->get_area(), !pane->is_in_vehicle() );
+                pane->index = 0;
+                // make sure to update the minimap as well!
+                refresh( recalc, true );
+                pane->needs_recalc = true;
+            }
+        } else {
+            popup( _( "No vehicle there!" ) );
+            refresh( recalc, true );
+        }
+    } else if( action == "CONSUME_FOOD" ) {
+        player &p = g->u;
+        item *it = sitem->items.front();
+        int pos = p.get_item_position( it );
+        if( pos != INT_MIN ) {
+            p.consume( pos );
+        } else if( p.consume_item( *it ) ) {
+            if( it->is_food_container() ) {
+                it->contents.erase( it->contents.begin() );
+                add_msg( _( "You leave the empty %s." ), it->tname() );
+            } else {
+                tripoint target = p.pos() + sitem->area->offset;
+                item_location loc;
+                if( sitem->from_vehicle ) {
+                    const cata::optional<vpart_reference> vp = g->m.veh_at( target ).part_with_feature( "CARGO", true );
+                    if( !vp ) {
+                        add_msg( _( "~~~~~~~~~ not vehicle?" ) );
+                        return action;
+                    }
+                    vehicle *const veh = &vp->vehicle();
```

Comment:
```suggestion
                    vehicle &veh = vp->vehicle();
```
So you don't need to dereference it below.